src/main/java/core/basesyntax/db/StockDaoStorageImpl.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.db;
+
+import core.basesyntax.storage.Storage;
+import java.util.Set;
+
+public class StockDaoStorageImpl implements StockDao {
+
+    @Override
+    public void set(String product, Integer amount) {
+        Storage.stock.put(product, amount);
+    }
+
+    @Override
+    public Integer get(String product) {
+        return Storage.stock.get(product);
+    }
+
+    @Override
+    public void decrease(String product, Integer amount) {
+        if (contain(product)) {
+            Storage.stock.put(product, get(product) - amount);
+        } else {
+            throw new RuntimeException("There was no such product before");
+        }
+    }
+
+    @Override
+    public void increase(String product, Integer amount) {
+        if (contain(product)) {
+            Storage.stock.put(product, get(product) + amount);

```

Comment:
Ensure that the increase operation does not lead to integer overflow. Although unlikely with typical inventory amounts, it's a good practice to handle such cases.